src/nghttp.cc

```diff
@@ -414,6 +482,14 @@ int submit_request(HttpClient *client, const Headers &headers, Request *req) {
 
   req->req_nva = std::move(build_headers);
 
+  if (expect_continue) {
+    auto timer = std::make_shared<ContinueTimer>(client->loop, req);
+    timer->start();
+
+    req->continue_timer = timer;
+    client->continue_timers.push_back(timer);
```

Comment:
I think it may be possible for Request to own std::unique_ptr of ContinueTimer (or embed timer in Request).  Request destructor will stop timer.